clients/fides-js/src/fides.ts

```diff
@@ -249,6 +254,12 @@ async function init(this: FidesGlobal, providedConfig?: FidesConfig) {
         if (!notice) {
           return null;
         }
+        if (typeof value === "string") {
+          // eslint-disable-next-line no-param-reassign
+          value = transformUserPreferenceToBoolean(
+            value as UserConsentPreference,
+          );
+        }
```

Comment:
I don't _love_ that we're having to check for "string" in many places now. Is there a way we can centralize this check to a shared place at least? My main concern is that if we either refactor or add additional code in the future it's easy to miss of of these generic "string" checks

Reply:
One idea is having a shared const `isStringConsent` which checks if `typeof value === "string"`.

Reply:
Good suggestion. I'll make a util for this and add some detailed docs.